docs/api/cli/deploy.md

```diff
@@ -88,11 +88,23 @@ Deploy an agent service.
     default=False,
     help="Apply environment variable when loading service config.",
 )
+@click.option(
+    "--use-hardhat",
```

Comment:
We removed this a while ago. Why are we reintroducing it? We had a good reason to remove it

Reply:
So I'm not aware on why the usage of hardhat node inside the deployment was removed, but we did not have this flag before we were including the hardhat node by default in docker compose deployments or in the dev mode in case of kubernetes deployments (https://github.com/valory-xyz/open-autonomy/pull/809#discussion_r876012304). I suppose we removed this because we wanted to use external blockchains in production. But this lead to a very annoying dev experience where we had to run a separate hardhat node when we wanted to run a local deployment. 

This flag makes the inclusion of the hardhat node optional, I think we can change this to be strictly for local deployments only but I think there is a pretty good reason to introduce this functionality

Reply:
Ok. Then make it strictly for local only please.